### ISSUE TITLE
Code Problem

### ISSUE THREAD

**Issue by fxnb** [wmn7/Universal_Light#9]
Hello author. We found that the state dimension is (5,12,7) in the code file, but (8,8,8) inside the paper. Also, we found that the test road network is incomplete. Can the author update the code, or provide a link to the original code file?

**Comment by wmn7** [comment]
Hello,

Thank you for your interest in my work.

Regarding the discrepancy between the code dimensions and those mentioned in the paper, this is due to an upgrade in the simulation dependency library. As noted in [Universal-Light Issue #7](https://github.com/wmn7/Universal-Light/issues/7), **I initially used Aiolos, which is an internal library developed by SenseTime. Therefore, I developed TransSimHub as a replacement for Aiolos. All the current code has been adapted to work with this new library.**

This required adapting the original code to the new library, which involved making some modifications. For instance, in the initial version, I used (8,8,8) dimensions, which did not account for right turns. In the new version, I have expanded to 12 dimensions to include right turns for greater generality. 

Regarding updates to the code, as I am currently interning, I have limited time to dedicate to updates. Therefore, unless there are convergence issues, there will likely not be significant updates to this repository.

Thank you once again for your support and understanding.

**Comment by fxnb** [comment]
> Hello,
> 
> Thank you for your interest in my work.
> 
> Regarding the discrepancy between the code dimensions and those mentioned in the paper, this is due to an upgrade in the simulation dependency library. As noted in [Universal-Light Issue #7](https://github.com/wmn7/Universal-Light/issues/7), **I initially used Aiolos, which is an internal library developed by SenseTime. Therefore, I developed TransSimHub as a replacement for Aiolos. All the current code has been adapted to work with this new library.**
> 
> This required adapting the original code to the new library, which involved making some modifications. For instance, in the initial version, I used (8,8,8) dimensions, which did not account for right turns. In the new version, I have expanded to 12 dimensions to include right turns for greater generality.
> 
> Regarding updates to the code, as I am currently interning, I have limited time to dedicate to updates. Therefore, unless there are convergence issues, there will likely not be significant updates to this repository.
> 
> Thank you once again for your support and understanding.

Thank you very much for your reply, we are using your code to produce very good results, and thank you for your open source work to bring great help to our work!

**Comment by wmn7** [comment]
Thank you once again for your interest in our work. 

As TransSimHub is still under development, there may be changes to the interface. If you encounter any issues with the code not running, please feel free to open an issue, and I will update it as soon as possible.

If you find our work helpful, please consider citing it.
```
@article{wang2024unitsa,
  author={Wang, Maonan and Xiong, Xi and Kan, Yuheng and Xu, Chengcheng and Pun, Man-On},
  journal={IEEE Transactions on Vehicular Technology}, 
  title={{UniTSA}: A Universal Reinforcement Learning Framework for V2X Traffic Signal Control}, 
  year={2024},
  volume={73},
  number={10},
  pages={14354-14369},
  doi={10.1109/TVT.2024.3403879}
}

@inproceedings{wang2022adlight,
  title={{ADLight}: A Universal Approach of Traffic Signal Control with Augmented Data Using Reinforcement Learning},
  author={Wang, Maonan and Xu, Yutong and Xiong, Xi and Kan, Yuheng and Xu, Chengcheng and Pun, Man-On},
  booktitle={2023 Transportation Research Board Annual Meeting (102nd TRB)},
  year={2023},
}
```

P.S. In the fine-tuning process, the learning rate significantly impacts the results. It's important to use different learning rates for fine-tuning and training.

Thank you for your understanding and support!
